File: app/src/main/java/com/example/myappcontacts/di/contactslist/ContactsListComponent.java
package com.example.myappcontacts.di.contactslist;

import com.example.myappcontacts.presentation.contactslist.presenter.ContactsListPresenter;

import dagger.Subcomponent;

@Subcomponent(modules = {ContactsListModule.class})
@ContactsListScope
public interface ContactsListComponent {
    void inject(ContactsListPresenter contactsListPresenter);
}

File: app/src/main/java/com/example/myappcontacts/di/contactslist/ContactsListComponent.kt
package com.example.myappcontacts.di.contactslist

import com.example.myappcontacts.presentation.contactslist.presenter.ContactsListPresenter

import dagger.Subcomponent

@Subcomponent(modules = [ContactsListModule::class])
@ContactsListScope
interface ContactsListComponent {
    fun inject(contactsListPresenter: ContactsListPresenter)
}
